patches/minecraft/net/minecraft/entity/player/EntityPlayer.java.patch

```diff
@@ -533,6 +533,39 @@
 +    public java.util.Collection<ITextComponent> getPrefixes() { return this.prefixes; }
 +    public java.util.Collection<ITextComponent> getSuffixes() { return this.suffixes; }
 +
++    private final net.minecraftforge.items.IItemHandler playerMainHandler = new net.minecraftforge.items.wrapper.PlayerMainInvWrapper(field_71071_by);
++    private final net.minecraftforge.items.IItemHandler playerArmorHandler = new net.minecraftforge.items.wrapper.PlayerArmorInvWrapper(field_71071_by);
++    private final net.minecraftforge.items.IItemHandler playerOffhandHandler = new net.minecraftforge.items.wrapper.PlayerOffhandInvWrapper(field_71071_by);
++    private final net.minecraftforge.items.IItemHandler playerJoinedHandler = new net.minecraftforge.items.wrapper.PlayerInvWrapper(field_71071_by);
++
++    @Override
++    public <T> T getCapability(net.minecraftforge.common.capabilities.Capability<T> capability, net.minecraft.util.EnumFacing facing)
++    {
++        if (capability == net.minecraftforge.items.CapabilityItemHandler.ITEM_HANDLER_CAPABILITY)
++        {
++            if (facing == net.minecraft.util.EnumFacing.UP)
++            {
++                return (T) playerArmorHandler;
++            } else if (facing == net.minecraft.util.EnumFacing.DOWN)
++            {
++                return (T) playerMainHandler;
++            } else if (facing.func_176740_k().func_176722_c())
++            {
++                return (T) playerOffhandHandler;
++            } else
++            {
++                return (T) playerJoinedHandler;
++            }
```

Comment:
These seem like arbitrary ways to define things?
I'd suggest. top/bottom Main Inventory. Sides Armor {including offhand, as it's kinda like an armor slot}